language/test/analyze.test.js

```diff
@@ -13,6 +13,372 @@
+
+      assert.equal(sample.mocks.language.document.calledOnce, true);
+      assert.deepEqual(sample.mocks.language.document.firstCall.args, [{
+        content: text,
+        type: type,
+        language: language
+      }]);
+      assert.equal(sample.mocks.document.annotate.calledOnce, true);
+      assert.deepEqual(sample.mocks.document.annotate.firstCall.args.slice(0, -1), [{ syntax: true }]);
+      assert.equal(callback.calledOnce, true);
+      assert.deepEqual(callback.firstCall.args, [null, sample.mocks.syntax]);
+      assert.equal(console.log.calledOnce, true);
+      assert.deepEqual(console.log.firstCall.args, ['Done analyzing syntax']);
+    });
+
+    it('should handle error', function () {
+      var error = new Error('error');
+      var sample = getSample();
+      var callback = sinon.stub();
+      sample.mocks.document.annotate.yields(error);
+
+      sample.program.analyzeSyntax(text, {
+        type: type,
+        language: language
+      }, callback);
+
+      assert.equal(callback.calledOnce, true);
+      assert.deepEqual(callback.firstCall.args, [error]);
+    });
+  });
+
+  describe('analyzeSyntaxFromFile', function () {
+    it('should analyze syntax in text', function () {
+      var sample = getSample();
+      var callback = sinon.stub();
+
+      sample.program.analyzeSyntaxFromFile(bucketName, fileName, {
+        type: type,
+        language: language
+      }, callback);
+
+      assert.equal(sample.mocks.language.document.calledOnce, true);
+      assert.deepEqual(sample.mocks.language.document.firstCall.args, [{
+        content: sample.mocks.file,
+        type: type,
+        language: language
+      }]);
+      assert.equal(sample.mocks.document.annotate.calledOnce, true);
+      assert.deepEqual(sample.mocks.document.annotate.firstCall.args.slice(0, -1), [{ syntax: true }]);
+      assert.equal(callback.calledOnce, true);
+      assert.deepEqual(callback.firstCall.args, [null, sample.mocks.syntax]);
+      assert.equal(console.log.calledOnce, true);
+      assert.deepEqual(console.log.firstCall.args, ['Done analyzing syntax']);
+    });
+
+    it('should handle error', function () {
+      var error = new Error('error');
+      var sample = getSample();
+      var callback = sinon.stub();
+      sample.mocks.document.annotate.yields(error);
+
+      sample.program.analyzeSyntaxFromFile(bucketName, fileName, {
+        type: type,
+        language: language
+      }, callback);
+
+      assert.equal(callback.calledOnce, true);
+      assert.deepEqual(callback.firstCall.args, [error]);
+    });
+  });
+
+  describe('main', function () {
+    it('should call analyzeSentiment', function () {
+      var program = getSample().program;
+
+      sinon.stub(program, 'analyzeSentiment');
+      program.main(['sentiment', text]);
+      assert.equal(program.analyzeSentiment.calledOnce, true);
+      assert.deepEqual(program.analyzeSentiment.firstCall.args.slice(0, -1), [text, {
+        type: 'text',
```

Comment:
Nit: for brevity's sake, should we define `{ type: 'text', language: undefined }` as a var and reuse that?